tests/online_migration/sle12_online_migration/yast2_migration.pm

```diff
@@ -19,6 +19,19 @@ sub run {
     }
     else {
         select_console 'x11';
+        # check screenlock after zypper patch
+        wait_still_screen;
+        send_key "backspace";
+        if (check_screen('screenlock')) {
+            send_key "esc";
+            if (check_screen 'gnome-screenlock-password') {
+                type_password;
+                send_key "ret";
+            }
+        }
+        mouse_hide(1);
+        assert_screen 'generic-desktop';
```

Comment:
you are duplicating content from `lib/utils.pm::wait_boot` and `tests/installation/first_boot.pm`.

[DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself). You are calling the same code multiple times, please [extract method](http://refactoring.com/catalog/extractMethod.html).